app/main.py

```diff
@@ -1,13 +1,23 @@
 def format_linter_error(error: dict) -> dict:
-    # write your code here
-    pass
+    return {
+        "line": error["line_number"],
+        "column": error["column_number"],
+        "message": error["text"],
+        "name": error["code"],
+        "source": "flake8"
+    }
 
 
 def format_single_linter_file(file_path: str, errors: list) -> dict:
-    # write your code here
-    pass
+    return {
+        "errors": [format_linter_error(error) for error in errors],
+        "path": file_path,
+        "status": "passed" if len(errors) == 0 else "failed"
+    }
 
 
 def format_linter_report(linter_report: dict) -> list:
-    # write your code here
-    pass
+    return [
+        format_single_linter_file(key_element, value_element)
+        for key_element, value_element in linter_report.items()
```

Comment:
key_element, value_element is not clear names

Reply:
Hi Arsen. This story for name of variables need to be close because it was took a lot of time. I don't understand what you mean "not clear"? I pushed new commit, if it's still wrong: can you explain to me what you mean "not clear" and some suggestion about it.
Best, Vasyl.